docs/core/advanced_tutorials/using-results.md

```diff
@@ -0,0 +1,226 @@
+
+Why would you persist the output of your tasks? The most common situation is to enable retrying from failure of a task for users that use a state database, such as Prefect Core's server or Prefect Cloud. By persisting the output of a task somewhere besides in-memory, a state database that knows the persisted location of a task's output can restart from where it left off. Another common situation is to allow for inspection of intermediary data when debugging flows (including -- maybe especially -- production ones). By checkpointing data during different data transformation Tasks, runtime data bugs can be tracked down by analyzing the checkpoints sequentially.
+
+## Setting up to handle results
+
+Checkpointing must be enabled for the pipeline to write a result. You must enable in two places: globally for the Prefect installation, and at the task level by providing a `Result` subclass to tasks either through its flow initialization or as a task-level override. Depending on which result subclass you configure, you may also need to ensure your authentication credentials are set up properly. For users of Prefect Core's server or Prefect Cloud, some of this is handled by default.
+
+For Core-only users, you must:
+
+- Opt-in to checkpointing globally by setting the `prefect.config.flows.checkpointing` to "True"
+- Specify the result your tasks will use for at least one level of specificity (flow-level or task-level)
+
+For Prefect Core server or Prefect Cloud users:
+
+- Checkpointing will automatically be turned on; you can disable it by passing `checkpoint=False` to each task you want to turn it off for
+- A result subclass that matches the storage backend of your `prefect.config.flows.storage` setting will automatically be applied to all tasks, if available; notably this is not yet supported for Docker Storage
+- You can override the automatic result subclass at the global level, flow level, or task level
+
+#### Setting result at the flow level
+```bash
+export PREFECT__FLOWS__CHECKPOINTING=true
+```
+
+```python{8-9}
+# flow.py
+from prefect.engine.results import LocalResult
+
+@task
+def add(x, y=1):
+    return x + y
+
+# send the configuration to the Flow object
+with Flow("my handled flow!", result=LocalResult()):
+    first_result = add(1, y=2)
+    second_result = add(x=first_result, y=100)
+```
+
+#### Setting result at the task level
+
+```bash
+export PREFECT__FLOWS__CHECKPOINTING=true
+```
+
+```python{4-5,13-14}
+# flow.py
+from prefect.engine.results import LocalResult
+
+# configure on the task decorator
+@task(result=LocalResult())
+def add(x, y=1):
+    return x + y
+
+class AddTask(Task):
+        def run(self, x, y):
+            return x + y
+
+# or when instantiating a Task object
+a = AddTask(result=LocalResult())
```

Comment:
Could we change a setting on the task result here to demonstrate the flexibility?  Or even use a different result class entirely, maybe `PrefectResult` ?

Reply:
Added more variety in 22d6508, though I avoided too much new config since the tutorial will explain templating piecemeal following this section